package/src/skia/web/JsiSkImage.ts

```diff
@@ -110,6 +122,60 @@ export class JsiSkImage extends HostObject<Image, "Image"> implements SkImage {
     return toBase64String(bytes);
   }
 
+  readPixels(
+    srcX?: number,
+    srcY?: number,
+    imageInfo?: ImageInfo,
+    dest?: MallocObj,
+    bytesPerRow?: number
+  ): Float32Array | Uint8Array | null {
+    const info = this.getImageInfo();
+    const pxInfo = {
+      colorSpace: this.CanvasKit.ColorSpace.SRGB,
+      width: imageInfo?.width ?? info.width,
+      height: imageInfo?.height ?? info.height,
+      colorType: ckEnum(
+        imageInfo?.colorType ?? this.CanvasKit.ColorType.RGBA_8888.value
+      ),
+      alphaType: ckEnum(imageInfo?.alphaType ?? info.alphaType),
+    };
+    if (typeof bytesPerRow !== "number") {
+      bytesPerRow = 4 * pxInfo.width;
+      if (pxInfo.colorType.value === this.CanvasKit.ColorType.RGBA_F16.value) {
+        bytesPerRow *= 2;
+      } else if (
+        pxInfo.colorType.value === this.CanvasKit.ColorType.RGBA_F32.value
+      ) {
+        bytesPerRow *= 4;
+      }
+    }
+    let destObj = dest;
+    if (!destObj) {
+      if (pxInfo.colorType.value === this.CanvasKit.ColorType.RGBA_F32.value) {
+        destObj = this.CanvasKit.Malloc(
```

Comment:
I don't think a CanvasKit.Malloc is necessary here.

Reply:
duplicated?